odoo/addons/base/models/image_mixin.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api, tools
+
+
+class ImageMixin(models.AbstractModel):
+    _name = 'image.mixin'
+    _description = "Image Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+
+    # all image_raw fields are technical and should not be displayed to the user
+    image_raw_original = fields.Binary("Raw Original Image", attachment=True)
+    image_raw_original_byte_size = fields.Integer("Byte Size of the Raw Original Image", compute='_compute_image_raw', store=True)
+
+    # resized fields stored (as attachment) for performance
+    image_raw_big = fields.Binary("Raw Big-sized Image", compute='_compute_image_raw', store=True)
+    image_raw_large = fields.Binary("Raw Large-sized Image", compute='_compute_image_raw', store=True)
+    image_raw_medium = fields.Binary("Raw Medium-sized Image", compute='_compute_image_raw', store=True)
+    image_raw_small = fields.Binary("Raw Small-sized Image", compute='_compute_image_raw', store=True)
+
+    # Computed fields that are used to create a fallback if necessary, it's
+    # recommended to display those fields to the user.
+    image_original = fields.Binary("Original Image", compute='_compute_image_original', inverse='_set_image_original', help="Image in its original size, as it was uploaded.")
```

Comment:
can we call it "image"  instead ???
sorry :/

Reply:
As discussed, image_big will be image.